apollo-client/src/main/java/com/ctrip/framework/apollo/internals/DefaultConfigManager.java

```diff
@@ -79,4 +79,8 @@ public ConfigFile getConfigFile(String namespace, ConfigFileFormat configFileFor
 
     return configFile;
   }
+
+  private String buildLock(String factor) {
+    return (String.format("%s.%s", getClass().getName(), factor)).intern();
```

Comment:
Using interned strings as locks does work, but it might not be the best way to go. Since the interned string is shared globally, locking the same string in different places could create a bit of a mix-up. So, why don't we stick with using concurrent hash maps to manage the locks for each namespace? It seems like a safer choice!

Reply:
> Using interned strings as locks does work, but it might not be the best way to go. Since the interned string is shared globally, locking the same string in different places could create a bit of a mix-up. So, why don't we stick with using concurrent hash maps to manage the locks for each namespace? It seems like a safer choice!

Yes, although it can work, this is not the safest way, may be we should perhaps sacrificing storage space for higher security.